### ISSUE TITLE
Selected vocabulary value is not saved 

### ISSUE THREAD

**Issue by mtrebron** [collective/collective.z3cform.datagridfield#134]
Given a TTW Dexterity type like 

```
<?xml version='1.0' encoding='utf8'?>
<model xmlns:i18n="http://xml.zope.org/namespaces/i18n" xmlns:form="http://namespaces.plone.org/supermodel/form" xmlns:security="http://namespaces.plone.org/supermodel/security" xmlns:marshal="http://namespaces.plone.org/supermodel/marshal" xmlns:lingua="http://namespaces.plone.org/supermodel/lingua" xmlns:users="http://namespaces.plone.org/supermodel/users" xmlns:indexer="http://namespaces.plone.org/supermodel/indexer" xmlns:easyform="http://namespaces.plone.org/supermodel/easyform" xmlns="http://namespaces.plone.org/supermodel/schema">
  <schema>
    <field name="dgf_choice" type="zope.schema.List">
        <description/>
        <required>False</required>
        <title>DGF Choice</title>
        <value_type type="collective.z3cform.datagridfield.DictRow">
        <schema>pnz.erpediem.core.interfaces.mvp.IDGFVocabTest</schema>
        </value_type>
        <form:widget type="collective.z3cform.datagridfield.datagridfield.DataGridFieldWidgetFactory"/>
    </field>
    <field name="vocab_choice" type="zope.schema.Choice">
      <description/>
      <required>False</required>
      <title>vocab choice</title>
      <vocabulary>plone.app.vocabularies.PortalTypes</vocabulary>
    </field>
  </schema>
</model>
```

with schema

```
class IDGFVocabTest(model.Schema):
    """ """
    my_vocab_choice = schema.Choice(
        title = _(
            'my_vocab_choice_title',
            default = u'Vocabulary Choice',
        ),
        vocabulary="plone.app.vocabularies.PortalTypes",
        required=False,
    )
    my_sanity_check = schema.TextLine(
        title = _(u'Textline'),
        required=False,
    )
```

![Screenshot from 2022-08-30 15-46-22](https://user-images.githubusercontent.com/14139595/187455254-210e20ae-4185-4e75-a44c-69e15d13d63b.png)

![Screenshot from 2022-08-30 15-46-36](https://user-images.githubusercontent.com/14139595/187455248-2156ace8-7170-4e85-875c-9a3c3cbb64e4.png)

![Screenshot from 2022-08-30 15-55-15](https://user-images.githubusercontent.com/14139595/187456000-5d160d54-2c14-42d5-aa29-a0bee649a364.png)

**Comment by petschki** [comment]
I had the same issue for `IRelatedItemsFieldWidget` and got it fixed with this PR: https://github.com/collective/collective.z3cform.datagridfield/pull/142 ... can you give this a try?

**Comment by mtrebron** [comment]
Sorry no difference with the new version. Downloaded fix-choicecolumn, reran pip, uninstalled and reinstalled from control panel, created the content type, and created an instance of the content...

**Comment by petschki** [comment]
I've hat the same problem with some registry forms and came up with another PR here #150 ... this completes the `DictRowConverter` and fixed several issues with `Choice` or `RelationList/Choice` fields for me. Can you give this a try?

**Comment by laulaz** [comment]
@petschki I can confirm that your PR fixes the issue where the choice field value is not saved  👍

**Comment by laulaz** [comment]
Fixed by #150  👍

**Comment by mtrebron** [comment]
Confirmed fixed, thank you @petschki
